src/endpoints/sources.rs

```diff
@@ -48,6 +49,23 @@ impl Source {
             .into_iter()
             .find(|source| source.name().as_deref() == Some(name))
     }
+
+    /// Create a source from it's unique id.
+    pub fn from_unique_id(unique_id: u32) -> Option<Source> {
```

Comment:
Can you please move the logic of this method into a method in `Sources` as

```rust
impl Sources {
    fn find_by_unique_id(unique_id: u32) -> Option<Source> { ... }
}
```

and then the `Source::from_unique_id` will just use `Sources::find_by_unique_id`.

Reply:
Done.